packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -157,20 +188,44 @@ class VisualStudio {
     return null;
   }
 
+  /// Checks if the given installation has issues that the user must resolve.
+  bool installationHasIssues(Map<String, dynamic>installationDetails) {
+    assert(installationDetails != null);
+    assert(installationDetails[_isCompleteKey] != null);
+    assert(installationDetails[_isRebootRequiredKey] != null);
+    assert(installationDetails[_isLaunchableKey] != null);
+
+    return installationDetails[_isCompleteKey] == false ||
+      installationDetails[_isRebootRequiredKey] == true ||
+      installationDetails[_isLaunchableKey] == false;
+  }
+
   /// Returns the details dictionary for the latest version of Visual Studio
   /// that has all required components.
   Map<String, dynamic> _cachedUsableVisualStudioDetails;
   Map<String, dynamic> get _usableVisualStudioDetails {
     _cachedUsableVisualStudioDetails ??=
         _visualStudioDetails(requiredComponents: _requiredComponents().keys);
+    // If a stable version is not found, try searching for a pre-release version.
+    _cachedUsableVisualStudioDetails ??= _visualStudioDetails(
+        requiredComponents: _requiredComponents().keys,
+        additionalArguments: <String>[_prereleaseKey]);
+    if (_cachedUsableVisualStudioDetails != null) {
+      if (installationHasIssues(_cachedUsableVisualStudioDetails)) {
+        _cachedAnyVisualStudioDetails = _cachedUsableVisualStudioDetails;
+        return null;
```

Comment:
It occurs to me reviewing this with more distance from having written it that the use of `null` to indicate both not having found anything usable and not having checked means we're going to run `vswhere` a bunch of extra times. It's a pre-existing problem which this change will make worse.

Could you do a follow-up PR to change the structure of these functions to use a sentinel value (maybe `{}`) for `_cached*` in cases where we have checked and not found anything?

Reply:
Here! https://github.com/flutter/flutter/pull/40197